rust-tdshim/build.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+
+use std::{
+    env, format, fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use rust_td_layout::build_time;
+
+fn nasm(file: &Path, arch: &str, out_file: &Path, args: &[&str]) -> Command {
+    let oformat = match arch {
+        "x86_64" => ("win64"),
+        "x86" => ("win32"),
+        "bin" => ("bin"),
+        _ => panic!("unsupported arch: {}", arch),
+    };
+    let mut c = Command::new("nasm");
+    let _ = c
+        .arg("-o")
+        .arg(out_file.to_str().expect("Invalid path"))
+        .arg("-f")
+        .arg(oformat)
+        .arg(file);
+    for arg in args {
+        let _ = c.arg(*arg);
+    }
+    c
+}
+
+fn run_command(mut cmd: Command) {
+    eprintln!("running {:?}", cmd);
+    let status = cmd.status().unwrap_or_else(|e| {
+        panic!("failed to execute [{:?}]: {}", cmd, e);
+    });
+    if !status.success() {
+        panic!("execution failed");
+    }
+}
+
+fn main() {
+    // tell cargo when to re-run the script
+    println!("cargo:rerun-if-changed=build.rs");
+    println!(
+        "cargo:rerun-if-changed={}",
+        Path::new("ResetVector/ResetVector.asm").to_str().unwrap()
+    );
+
+    let old_current_dir = env::current_dir().unwrap();
+    let new_current_dir = old_current_dir.join("ResetVector");
+    let out_dir = env::var("OUT_DIR").unwrap();
+    let out_file = PathBuf::from(out_dir).join("ResetVector.bin");
+    let copy_to_dir = out_file
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap();
+    let copy_to_file = copy_to_dir.join("ResetVector.bin");
+
+    eprintln!("out_file is     {}", out_file.to_str().unwrap());
+    eprintln!("copy_to_file is {}", copy_to_file.to_str().unwrap());
```

Comment:
The purpose is to output no matter success or fail, so that we can dump the info from build log.

Is there any concern?

Reply:
@jyao1 - afaik a build script will only produce output when it fails (`exit(1)`) *unless* you use the "magic" `cargo:warning` prefix:

```rust
eprintln!("cargo:warning=out_file is     {}", out_file.to_str().unwrap());
```

See: https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargo-warning

Reply:
Right. That is what we expect. See below log:
```
Compiling rust-td-payload v0.1.0 (/home/luxy/code/jyao1/td-shim/rust-td-payload)
warning: reset_vector_bin_file is /home/luxy/code/jyao1/td-shim/target/x86_64-unknown-uefi/release/build/rust-tdshim-5832cce1ce4b1df4/out/../../../ResetVector.bin
Finished release [optimized] target(s) in 2.48s
```

Because ResetVector.bin is generated by the build script instead of normal rust build, we hope to see the final path of ResetVector.bin so that we can easily known where it is.
That information is very helpful when we debug

Reply:
Thanks for the reminder. We revisit and realize that it does not need to be print.
The ResetVector.bin is put to the same location as rust-tdshim.

We will remove it here.